app/controllers/system_notifications_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class SystemNotificationsController < ApplicationController
+  def index
+    @system_notifications = prepare_notifications
+    respond_to do |format|
+      format.json do
+        render json: {
+          more_url: prepare_notifications.fetch(:more_notifications_url),
```

Comment:
Replace the duplicated call to `prepare_notifications` with `@system_notifications`.